models/progresses.js

```diff
@@ -0,0 +1,138 @@
+const { Conflict } = require("http-errors");
+const fireStore = require("../utils/firestore");
+const progressesCollection = fireStore.collection("progresses");
+const { fetchTask } = require("./tasks");
+const { fetchUser } = require("./users");
+
+/**
+ * Adds a new progress document for the given user or task, with a limit of one progress document per day.
+ * @param progressData {object} The data to be added. It should be an object containing key-value pairs of the fields to be added, including a "type" field set to either "user" or "task".
+ * @returns {Promise<object>} A Promise that resolves with the added progress document object, or rejects with an error object if the add operation fails.
+ * @throws {Error} If a progress document has already been created for the given user or task on the current day.
+ **/
+const createProgressDocument = async (progressData) => {
+  const { type, userId, taskId } = progressData;
+  const createdAtTimestamp = new Date().getTime();
+  const currentHourIST = new Date().getUTCHours() + 5.5; // IST offset is UTC+5:30
+  const isBefore6amIST = currentHourIST < 6;
+  const progressDateTimestamp = isBefore6amIST
+    ? new Date().setUTCHours(0, 0, 0, 0) - 24 * 60 * 60 * 1000
```

Comment:
Should this `24 * 60 ...` be moved into a variable as it's not very descriptive at first glance of what this is?